pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/generic/GenericJsonWriter.java

```diff
@@ -36,7 +38,11 @@ public GenericJsonWriter() {
     @Override
     public byte[] write(GenericRecord message) {
         try {
-            return objectMapper.writeValueAsBytes(((GenericJsonRecord)message).getJsonNode().toString());
+            ByteArrayOutputStream buffer = new ByteArrayOutputStream();
```

Comment:
I wonder if there's any specific reason we want to create our own ByteArrayOutputStream & JsonGenerator, which is what [objectMapper.writeValueAsBytes()](https://github.com/FasterXML/jackson-databind/blob/7ccdb596ce6ce3389aa5b9824cab70441fd72322/src/main/java/com/fasterxml/jackson/databind/ObjectMapper.java#L1699-L1708) is doing internally, instead it is using a ByteArrayBuilder which is basically Jackson's implementation of OutputStream and claim to be more efficient in many of its use case.
I think we should just use objectMapper.writeValueAsBytes() unless we want to write output to somewhere else(not a byte[]).

Reply:
+1 to what @MarvinCai said

Reply:
@MarvinCai thanks for the heads up. I was overdoing it. :) I pushed a fix where I simply dropped `.toString()`, as you had already pointed out in the issue report.